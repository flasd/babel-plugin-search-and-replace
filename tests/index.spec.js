const babel = require("@babel/core");
const searchAndReplaceplugin = require("../lib/index");

var example = `
var foo = 1;
if (foo) console.log(foo);
const hello = "hello";

const myFunc = function(param = "hello"){
    return 1;
}
`;

it("should replace nothing", () => {
  const { code } = babel.transform(example, {
    plugins: [searchAndReplaceplugin]
  });
  expect(code).toMatchSnapshot();
});

it("shouldn't replace foo by bar", () => {
  const { code } = babel.transform(example, {
    plugins: [
      [
        searchAndReplaceplugin,
        {
          rules: [
            {
              search: "foo",
              replace: "bar"
            }
          ]
        }
      ]
    ]
  });
  expect(code).toMatchSnapshot();
});

it("should replace hello by hey", () => {
  const { code } = babel.transform(example, {
    plugins: [
      [
        searchAndReplaceplugin,
        {
          rules: [
            {
              search: "foo",
              replace: "bar"
            },
            {
              search: "hello",
              replace: "hey"
            }
          ]
        }
      ]
    ]
  });
  expect(code).toMatchSnapshot();
});

it("should replace el by aze", () => {
  const { code } = babel.transform(example, {
    plugins: [
      [
        searchAndReplaceplugin,
        {
          rules: [
            {
              search: /el/,
              replace: "aze"
            }
          ]
        }
      ]
    ]
  });
  expect(code).toMatchSnapshot();
});


var exampleWithTemplates = `
var foo = 1;
if (foo) console.log(foo);
const bye = \`bye\${foo}\`;

const myFunc = function(param = \`bye\${foo}\`){
    return 1;
}
`;

it("should replace bye for adios", () => {
  const { code } = babel.transform(exampleWithTemplates, {
    plugins: [
      [
        searchAndReplaceplugin,
        {
          rules: [
            {
              search: 'bye',
              searchTemplateStrings: true,
              replace: "adios",
            }
          ]
        }
      ]
    ]
  });
  expect(code).toMatchSnapshot();
});

it("should replace ye by aby", () => {
  const { code } = babel.transform(exampleWithTemplates, {
    plugins: [
      [
        searchAndReplaceplugin,
        {
          rules: [
            {
              search: /ye/,
              searchTemplateStrings: true,
              replace: "aby",
            }
          ]
        }
      ]
    ]
  });
  expect(code).toMatchSnapshot();
});