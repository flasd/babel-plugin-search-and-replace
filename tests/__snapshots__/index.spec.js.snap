// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should replace bye for adios 1`] = `
"var foo = 1;
if (foo) console.log(foo);
const bye = \`adios\${foo}\`;

const myFunc = function (param = \`adios\${foo}\`) {
  return 1;
};"
`;

exports[`should replace el by aze 1`] = `
"var foo = 1;
if (foo) console.log(foo);
const hello = \\"hazelo\\";

const myFunc = function (param = \\"hazelo\\") {
  return 1;
};"
`;

exports[`should replace hello by hey 1`] = `
"var foo = 1;
if (foo) console.log(foo);
const hello = \\"hey\\";

const myFunc = function (param = \\"hey\\") {
  return 1;
};"
`;

exports[`should replace nothing 1`] = `
"var foo = 1;
if (foo) console.log(foo);
const hello = \\"hello\\";

const myFunc = function (param = \\"hello\\") {
  return 1;
};"
`;

exports[`should replace ye by aby 1`] = `
"var foo = 1;
if (foo) console.log(foo);
const bye = \`baby\${foo}\`;

const myFunc = function (param = \`baby\${foo}\`) {
  return 1;
};"
`;

exports[`shouldn't replace foo by bar 1`] = `
"var foo = 1;
if (foo) console.log(foo);
const hello = \\"hello\\";

const myFunc = function (param = \\"hello\\") {
  return 1;
};"
`;
